src/Umbraco.Cms.Api.Management/Controllers/Document/ByKeyDocumentController.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Umbraco.Cms.Api.Management.Factories;
+using Umbraco.Cms.Api.Management.ViewModels.Document;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Services;
+
+namespace Umbraco.Cms.Api.Management.Controllers.Document;
+
+public class ByKeyDocumentController : DocumentControllerBase
+{
+    private readonly IContentService _contentService;
+    private readonly IDocumentViewModelFactory _documentViewModelFactory;
+
+    public ByKeyDocumentController(IContentService contentService, IDocumentViewModelFactory documentViewModelFactory)
+    {
+        _contentService = contentService;
+        _documentViewModelFactory = documentViewModelFactory;
+    }
+
+    [HttpGet("{key:guid}")]
+    [MapToApiVersion("1.0")]
+    [ProducesResponseType(typeof(DocumentViewModel), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<DocumentViewModel>> ByKey(Guid key)
+    {
+        IContent? content = _contentService.GetById(key);
```

Comment:
Wonder if it is better to hide this behind a private method returning attempt. Thereby we can start the pattern and use the same method in the base for Id not found, as we will need for Update operations.

Reply:
We haven't used the attempt pattern for simply retrieving items anywhere else; but we do have async Get-methods in the new service implementations. We should do something similiar.

The create/update PR will include a new service for content editing, which is using async + attempt. I'll add a FIXME here and include something in the create/update service for retrieving content items async.